be/src/Project/Web/Buyer/Buyer.Web/Shared/ModelBuilders/MainNavigations/MainNavigationModelBuilder.cs

```diff
@@ -4,91 +4,40 @@
-        private readonly LinkGenerator linkGenerator;
-        private readonly IOptionsMonitor<AppSettings> settings;
+        private readonly IOptionsMonitor<AppSettings> _settings;
+        private readonly IMainNavigationLinkRepository _mainNavigationLinkRepository;
 
         public MainNavigationModelBuilder(
-            IStringLocalizer<GlobalResources> globalLocalizer,
-            IStringLocalizer<OrderResources> orderLocalizer,
-            LinkGenerator linkGenerator,
-            IOptionsMonitor<AppSettings> settings)
+            IOptionsMonitor<AppSettings> settings,
+            IMainNavigationLinkRepository mainNavigationLinkRepository)
         {
-            this.globalLocalizer = globalLocalizer;
-            this.orderLocalizer = orderLocalizer;
-            this.linkGenerator = linkGenerator;
-            this.settings = settings;
+            _settings = settings;
+            _mainNavigationLinkRepository = mainNavigationLinkRepository;
         }
 
         public async Task<MainNavigationViewModel> BuildModelAsync(ComponentModelBase componentModel)
         {
-            var links = new List<LinkViewModel>
-            { 
-                new LinkViewModel 
-                {
-                    Text = this.globalLocalizer.GetString("Home"),
-                    Url = this.linkGenerator.GetPathByAction("Index", "Home", new { Area = "Home", culture = CultureInfo.CurrentUICulture.Name })
-                }
-            };
-
-            links.Add(new LinkViewModel
-            {
-                Text = this.orderLocalizer.GetString("MyOrders").Value,
-                Url = this.linkGenerator.GetPathByAction("Index", "Orders", new { Area = "Orders", culture = CultureInfo.CurrentUICulture.Name })
-            });
-
-            links.Add(new LinkViewModel
-            {
-                Text = this.orderLocalizer.GetString("PlaceOrder").Value,
-                Url = this.linkGenerator.GetPathByAction("Index", "Order", new { Area = "Orders", culture = CultureInfo.CurrentUICulture.Name })
-            });
-
-            links.Add(new LinkViewModel
-            {
-                Text = this.globalLocalizer.GetString("AvailableProducts"),
-                Url = this.linkGenerator.GetPathByAction("Index", "AvailableProducts", new { Area = "Products", culture = CultureInfo.CurrentUICulture.Name })
-            });
-
-            links.Add(new LinkViewModel
-            {
-                Text = this.globalLocalizer.GetString("Outlet").Value,
-                Url = this.linkGenerator.GetPathByAction("Index", "Outlet", new { Area = "Products", culture = CultureInfo.CurrentUICulture.Name })
-            });
-
-            links.Add(new LinkViewModel
-            {
-                Text = this.globalLocalizer.GetString("News"),
-                Url = this.linkGenerator.GetPathByAction("Index", "News", new { Area = "News", culture = CultureInfo.CurrentUICulture.Name })
-            });
-
-            links.Add(new LinkViewModel
```

Comment:
**suggestion:** Please leave the links up to Download Center as they were and add the rest from Strapi